### ISSUE TITLE
How to modify your code LapSRN_WGAN for  same size image  processing?

### ISSUE THREAD

**Issue by kirill-pinigin** [twtygqyy/pytorch-LapSRN#8]
Hello. Please help me how to modify your code LapSRN_WGAN for  same size image  processing without upsampling input image? I want use LapSRN_WGAN for image processing with same size

Should I modify this line of code in https://github.com/twtygqyy/pytorch-LapSRN/blob/master/lapsrn_wgan.py

Change " nn.ConvTranspose2d(in_channels=64, out_channels=64, kernel_size=4, stride=2, padding=1, bias=False)",  to nn.Conv2d(in_channels=64, out_channels=64, kernel_size=4, stride=1, padding=1, bias=False), 

Or may be I can just remove this layer?

**Comment by twtygqyy** [comment]
Hi @kirill-pinigin, you are write, but you will have to change the size of the input training samples. Check my VDSR repo which same size input and output are used for training.

**Comment by kirill-pinigin** [comment]
Thank you I understood. I chahge size of output image. But I don't understand what I must do with  nn.ConvTranspose2d layer? Just remove or replace  on simple Conv2D?

**Comment by twtygqyy** [comment]
@kirill-pinigin you can just remove nn.ConvTranspose2d layers.

**Comment by kirill-pinigin** [comment]
May I ask you? If you do not mind I need to know your expert opinion.  Can I use LapSRN_WGAN for image inpainting? I want to eliminate  artefacts from  grayscale image?  What should I do? Can LapSRN_WGAN reduce or even elminate  scratches or other artefacts from grayscale photo? What can you advise me?

**Comment by twtygqyy** [comment]
Hi @kirill-pinigin Of course the code can be modified for image in-painting. Basically there are two steps you need to do:
1. Remove deconvolutional layers, keep input and output in same shape. You can check my VDSR repo for reference.
2. Create input and output training datasets. I guess input could be image with mask, and output should be the raw image.

**Comment by kirill-pinigin** [comment]
I have  already used your VDSR model in same maner like iitem 2   .  unfortunately qualiy of restored images is not appropriate. May you advise some method to improve perceptual quality  of  restored images. 
![bad_samples_230](https://user-images.githubusercontent.com/27665392/36371813-d4ec80a6-1595-11e8-83ff-c1bddb986703.png)
![reconst_230_psnr 30 8107676076](https://user-images.githubusercontent.com/27665392/36371815-d66a6f10-1595-11e8-8436-d7f1a6519868.png)

**Comment by twtygqyy** [comment]
@kirill-pinigin I will suggest you to use GAN instead of MSE-loss for training the network. It usually works better for this task. Check out http://hi.cs.waseda.ac.jp/~iizuka/projects/completion/en/
